src/Orleans.CodeGenerator/LibraryTypes.cs

```diff
@@ -237,6 +251,31 @@ INamedTypeSymbol TypeOrDefault(string metadataName)
         private INamedTypeSymbol IPEndPoint;
         private INamedTypeSymbol CancellationToken;
         private INamedTypeSymbol[] ImmutableContainerTypes;
+        private INamedTypeSymbol Guid;
+        private INamedTypeSymbol BitVector32;
+        private INamedTypeSymbol CompareInfo;
+        private INamedTypeSymbol CultureInfo;
+        private INamedTypeSymbol Version;
+        private INamedTypeSymbol[] _regularShallowCopyableTypes;
+        private INamedTypeSymbol[] RegularShallowCopyableType => _regularShallowCopyableTypes ??= new[]
+        {
+            TimeSpan,
+            DateOnly,
+            TimeOnly,
+            DateTimeOffset,
+            Guid,
+            BitVector32,
+            CompareInfo,
+            CultureInfo,
+            Version,
+            IPAddress,
+            IPEndPoint,
+            CancellationToken,
+            Type,
+            Uri
```

Comment:
Note that these are the "regular" types which are considered immutable. There is also a set of "special" immutable types:
```csharp
                case SpecialType.System_Boolean:
                case SpecialType.System_Char:
                case SpecialType.System_SByte:
                case SpecialType.System_Byte:
                case SpecialType.System_Int16:
                case SpecialType.System_UInt16:
                case SpecialType.System_Int32:
                case SpecialType.System_UInt32:
                case SpecialType.System_Int64:
                case SpecialType.System_UInt64:
                case SpecialType.System_Decimal:
                case SpecialType.System_Single:
                case SpecialType.System_Double:
                case SpecialType.System_String:
                case SpecialType.System_DateTime:
```